news/_posts/2022-07-01-introducing-horizontal-migration.md

```diff
@@ -0,0 +1,74 @@
+
+Moreover, every bit of the legacy framework must be preserved until the migration has been fully completed. Even if in a few years only a single page that few people use was left to migrate, everything it relies on, Dispatcher, ObjectModel, Smarty, Helpers... the entire legacy stack must remain in place, alive and maintained.
+
+Of course, in time, the migration will be finished, we will be able to remove any remaining legacy components, and reduce the platform's complexity. But as the saying goes, "in the long-term, we will all be dead". PrestaShop's attractiveness as a platform is at risk if we don't manage to accelerate the migration.
+
+**We need a better strategy.** How can we accelerate the transition to Symfony, increase the platform's homogeneity, and phase out legacy components faster?
+
+## Thinking outside the box
+
+Let's analyze our current strategy. Migrating a page takes a list of known steps. Currently, we perform all those steps on a single page, then move on to the next page, and so on. 
+
+What if we focused on delivering a single step at a time, progressively across all pages? 
+
+Take the controller part, for example. At its root, a Symfony controller takes a Request and provides a Response, that's all. How the request is processed in order to build that response has no impact on the nature of the controller. Who says Symfony controller must use Twig? Or Grid? Or Symfony forms, even? Theoretically, it should be possible to take the content of the legacy controller and paste it into a Symfony controller. As long as the original code is rearranged into actions, and we get a response, it doesn't matter if the output is built using Smarty helpers: HTML is HTML.
+
+Following that logic, we could say that the first step to migrating a controller is "transforming" it into a Symfony controller. Formally speaking, this requires minimal changes to the original code. It might not be pretty, but it provides all the benefits of a full-fledged Symfony controller (routing, kernel events, debug features, translator, container, etc) – for a low, low price.
+
+Migrating a page using this "shortcut" should take fewer modifications than doing it the traditional way. This means less developer effort (making it easier for contributors to participate), less effort to review and, hopefully, less effort to verify. If we did this to all remaining controllers, we could have 100% of the Back office running on Symfony controllers much sooner than if we performed all the refactoring steps to each of them as we do now. This would allow us to phase out Dispatcher, Link, AdminController, and other legacy components much faster.
```

Comment:
Alternative version:

I kept the parentheses because they relate to the "developer effort" -> less effort means it's easier for contributors to do it. Perhaps there's a better way.

I also insist on keeping "If we did" because I feel it more closely represents my intention. At this point in my discourse, I'm trying to suggest a potential course of action, not a direct cause-effect correlation.

```suggestion
Migrating a page using this "shortcut" should take fewer modifications compared to the traditional way. This means less developer effort (making it easier for contributors to participate), less effort to review and, hopefully, less effort to verify. If we did this to all remaining controllers, we could have 100% of the back office running on Symfony controllers much sooner than if we performed all the refactoring steps to each of them as we do now. This would allow us to phase out Dispatcher, Link, AdminController, and other legacy components much faster.
```